test-e2e/utils.js

```diff
@@ -194,21 +194,28 @@ export async function createManagers(count, t, deviceType) {
  * @param {import('../src/generated/rpc.js').DeviceInfo['deviceType']} [deviceType]
  */
 export function createManager(seed, t, deviceType) {
-  const dbFolder = FAST_TESTS ? ':memory:' : temporaryDirectory()
-  const coreStorage = FAST_TESTS ? () => new RAM() : temporaryDirectory()
-
-  t.teardown(async () => {
-    if (FAST_TESTS) return
-    await Promise.all([
-      fsPromises.rm(dbFolder, { recursive: true, force: true, maxRetries: 2 }),
-      // @ts-ignore
-      fsPromises.rm(coreStorage, {
-        recursive: true,
-        force: true,
-        maxRetries: 2,
-      }),
-    ])
-  })
+  /** @type {string} */ let dbFolder
+  /** @type {string | import('../src/types.js').CoreStorage} */ let coreStorage
+
+  if (FAST_TESTS) {
+    dbFolder = ':memory:'
+    coreStorage = () => new RAM()
+  } else {
+    const directories = [temporaryDirectory(), temporaryDirectory()]
+    ;[dbFolder, coreStorage] = directories
+    t.teardown(() =>
+      Promise.all(
+        directories.map((dir) =>
```

Comment:
countering your nitpick with my own nitpick 😄 (i always find the semi-colon trick(?) a little hard to parse)

```suggestion
    dbFolder = temporaryDirectory()
    coreStorage = temporaryDirectory()
    t.teardown(() =>
      Promise.all(
        [dbFolder, coreStorage].map((dir) =>
```

Reply:
This doesn't work because TypeScript doesn't know that `coreStorage` is a string at this point. We could fix that by defining an intermediate variable?

Reply:
ah welp nevermind 😂  - probably not worth going out of your way to change what you had before, especially since this is just test-specific